src/main/java/com/adobe/target/edge/client/model/TargetAttributesResponse.java

```diff
@@ -84,80 +115,99 @@ public TargetDeliveryResponse getResponse() {
-            String mbox = resp.getName();
-            List<Option> options = resp.getOptions();
-            addOptions(options, mbox);
-          }
-        }
-      }
-      ExecuteResponse executeResponse = response.getExecute();
-      if (executeResponse != null) {
-        PageLoadResponse pageloadResponse = executeResponse.getPageLoad();
-        if (pageloadResponse != null) {
-          addOptions(pageloadResponse.getOptions(), globalMbox);
-        }
-        List<MboxResponse> mboxes = executeResponse.getMboxes();
-        if (mboxes != null) {
-          for (int i = mboxes.size() - 1; i >= 0; i--) {
-            MboxResponse resp = mboxes.get(i);
-            String mbox = resp.getName();
-            List<Option> options = resp.getOptions();
-            addOptions(options, mbox);
-          }
-        }
-      }
-      this.contentCreated = true;
+
+    processPageLoad(accumulator, response.getPageLoad(), globalMbox);
+    processMboxes(accumulator, response.getMboxes());
+  }
+
+  private static void processExecute(
+      Map<String, Map<String, Object>> accumulator, ExecuteResponse response, String globalMbox) {
+    if (response == null) {
+      return;
     }
-    return this.content;
+
+    processPageLoad(accumulator, response.getPageLoad(), globalMbox);
+    processMboxes(accumulator, response.getMboxes());
+  }
+
+  private static void processPageLoad(
+      Map<String, Map<String, Object>> accumulator, PageLoadResponse response, String globalMbox) {
+    if (response == null) {
+      return;
+    }
+
+    addOptions(accumulator, response.getOptions(), globalMbox);
   }
 
-  private void addOptions(List<Option> options, String mbox) {
-    Map<String, Object> mboxContent = this.content.computeIfAbsent(mbox, k -> new HashMap<>());
+  private static <T extends MboxResponse> void processMboxes(
+      Map<String, Map<String, Object>> accumulator, List<T> mboxes) {
+    if (mboxes == null) {
+      return;
+    }
+
+    for (int i = mboxes.size() - 1; i >= 0; i--) {
+      MboxResponse resp = mboxes.get(i);
+      String mbox = resp.getName();
+      List<Option> options = resp.getOptions();
+
+      addOptions(accumulator, options, mbox);
+    }
+  }
+
+  private static void addOptions(
+      Map<String, Map<String, Object>> accumulator, List<Option> options, String mbox) {
+    Map<String, Object> mboxContent = accumulator.computeIfAbsent(mbox, k -> new HashMap<>());
+
     for (int j = options.size() - 1; j >= 0; j--) {
       Option option = options.get(j);
-      Object contentMap = option.getContent();
-      if (option.getType() == OptionType.JSON && contentMap instanceof Map) {
+
+      if (isJsonOption(option)) {
+        Object contentMap = option.getContent();
         @SuppressWarnings("unchecked")
         Map<String, Object> contentObj = (Map<String, Object>) contentMap;
         mboxContent.putAll(contentObj);
       }
     }
   }
+
+  private static boolean isJsonOption(Option option) {
```

Comment:
its not obvious to me why this was broken out of the addOptions function... is it an optimization of some sort?

Reply:
@donwalling  just for readability sake.